src/App.tsx

```diff
@@ -1,97 +1,80 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPer, setItemsPer] = useState(5);
+  const [activePage, setActivePage] = useState(1);
+  const [
+    visibleItems, setVisibleItems,
+  ] = useState<string[]>([]);
+
+  useEffect(() => {
+    const firstItem = itemsPer * (activePage - 1);
+    const nextPageItem = firstItem + itemsPer;
+
+    setVisibleItems(items.slice(firstItem, nextPageItem));
+  }, [itemsPer, activePage]);
+
+  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setItemsPer(+event.target.value);
+    setActivePage(1);
+  };
+
+  const handlePageChange = (page: number) => {
+    setActivePage(page);
+  };
+
+  const firstVisibleIndex = items.indexOf(visibleItems[0]) + 1;
+  const lastVisibleIndex = items
+    .indexOf(visibleItems[visibleItems.length - 1]) + 1;
+
   return (
-    <div className="container">
-      <h1>Items with Pagination</h1>
+    <>
+      <div className="container">
+        <h1>Items with Pagination</h1>
 
-      <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
-      </p>
+        <p className="lead" data-cy="info">
+          {`Page ${activePage} (items ${firstVisibleIndex} - ${lastVisibleIndex} of ${items.length})`}
+        </p>
 
-      <div className="form-group row">
-        <div className="col-3 col-sm-2 col-xl-1">
-          <select
-            data-cy="perPageSelector"
-            id="perPageSelector"
-            className="form-control"
-          >
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
-          </select>
-        </div>
+        <div className="form-group row">
+          <div className="col-3 col-sm-2 col-xl-1">
+            <select
+              data-cy="perPageSelector"
+              id="perPageSelector"
+              className="form-control"
+              value={itemsPer}
+              onChange={handleSelect}
+            >
+              <option value="3">3</option>
```

Comment:
Please store it somewhere as a constant and map here

Reply:
I mean store amount of elements per page as a collection above the component like 
`const itemsPerPage = [5, 10, 15] ...`